keras_cv/models/CLIP/clip_encoder.py

```diff
@@ -0,0 +1,262 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from keras_cv.backend import keras
+from keras_cv.backend import ops
+
+
+def get_initializer(initializer_range=0.02):
+    """
+    Creates a `keras.initializers.TruncatedNormal` with the given range.
+
+    Args:
+        initializer_range (*float*, defaults to 0.02): Standard deviation of the initializer range.
+
+    Returns:
+        `keras.initializers.TruncatedNormal`: The truncated normal initializer.
+    """
+    return keras.initializers.TruncatedNormal(stddev=initializer_range)
+
+
+class QuickGELU(keras.layers.Layer):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def call(self, x):
+        return x * ops.sigmoid(1.702 * x)
+
+
+class ResidualAttention(keras.layers.Layer):
+    def __init__(
+        self,
+        proj_dim,
+        n_head,
+        num_hidden_layers,
+        attn_mask=None,
+    ):
+        super().__init__()
+        self.proj_dim = proj_dim
+        self.n_head = n_head
+        self.attn_mask = attn_mask
+        self.num_hidden_layers = num_hidden_layers
+        self.fc_std = ops.power(2 * self.proj_dim, -0.5) * 0.02
+
+        self.in_proj_std = (
+            ops.power(self.proj_dim, -0.5)
+            * (ops.power(2 * self.num_hidden_layers, -0.5))
+            * 0.02
+        )
+
+    def attention(self, x):
+        self.attn_mask = (
+            ops.cast(self.attn_mask, dtype=x.dtype)
+            if self.attn_mask is not None
+            else None
+        )
+
+        return self.attn(x, attention_mask=self.attn_mask)
+
+    def build(self, input_shape):
+        super().build(input_shape)
```

Comment:
```
super().build(input_shape)
```
Is it necessary?

Also, the `input_shape` is not being used in the `build` method. So, why defining the layers here instead of `__init__`?

Reply:
sometimes the graph is not correctly traced and defining trainable variables in build makes sure there are no build state issues